clusterloader2/docs/test_suite_design.md

```diff
@@ -0,0 +1,42 @@
+# Cluster loader - TestSuite
+
+## Background
+
+Since 2019 cluster loader become main tool for kubernetes e2e performance tests.
+It is a very convenient tool for running single test case, but not for running
+test bundles. Currently it is impossible to run the same test
+with two different parameter sets in a single job run.
+This forces developers to create separate job configs for almost each test.
+
+## TestSuite
```

Comment:
Any chance to have an example of how tests will look like before and after the change. I am having troubles to understand the benefit of this change.

Reply:
This will allow us to have same test template multiple times in a run. For example:
```
- ConfigPath: storage_test.yaml
  OverridePaths:
  - use_storage_type1.yaml
- ConfigPath: storage_test.yaml
  OverridePaths:
  - use_storage_type2.yaml
- ConfigPath: storage_test.yaml
  OverridePaths:
  - use_storage_type3.yaml
```
Previously we had to create separate ci job for each test. IMHO the gains are:
- It's more or less that same thing we define we current cluster loader flags, however we will able to assign overrides to specific test.
- We can bundle test together, so we don't have to create and tear down cluster for each test case. For short tests this will be a huge gain.